File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxi = 0
        i = 0
        j = len(height) - 1
        while i < j:
            res = (j - i) * min(height[i],height[j])
            if res > maxi:
                maxi = res
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
            
        return maxi

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max = 0;
        int i = 0;
        int j = height.size() - 1;
        while(i < j){
            if(height[i] < height[j]){
                if (max < ((j - i) * (height[i])))
                    max = ((j - i) * (height[i]));
                i++;
            }
            else{
                if (max < ((j - i) * (height[j])))
                    max = ((j - i) * (height[j]));
                j--;
            }
        }
        return max;
    }
};